Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -1112,10 +1129,13 @@ public IEnumerable<TItem> Data
             get { return data; }
             set
             {
-                data = value;
+                //This was just straight up recalculating every grid interaction...
+                //Need to figure out a better way to track reloading filter/view
+                if ( value == null || ( !data?.SequenceEqual( value ) ?? true ) )

```

Comment:
I didn't know about `SequenceEqual` method. Is it the same thing as we already have, a `ArrayExtensions.AreEqual`?

Reply:
Oh I didn't even notice you had that extension, haha.
But yes it is equivalent, makes sure that for both sequences every item is compared by using the default comparer. Maybe you could actually replace it. I would think the native implementation might have slightly better performance.
Something we could compare by using, if needed:
https://benchmarkdotnet.org/articles/overview.html